### DISCUSSION TITLE
can it work under tf 2.x?

### DISCUSSION THREAD

**Discussion by lg2578** [Rishit-dagli/Gradient-Centralization-TensorFlow#26] [Show and tell]
1.get_gradients only support in tf 1 lengcy graph mode ,the code seems can't work under tf 2.x
2.I've runned the fashion-mnist demo, the model's compile method not init the model's weight, so  may need to create model again before run the gctf. 
like the following  italic code:
import gctf #import gctf

time_callback_gctf = TimeHistory()
_**model = tf.keras.models.Sequential([
                                    tf.keras.layers.Flatten(), 
                                    tf.keras.layers.Dense(512, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(256, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(64, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(512, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(256, activation=tf.nn.relu),
                                    tf.keras.layers.Dense(64, activation=tf.nn.relu), 
                                    tf.keras.layers.Dense(10, activation=tf.nn.softmax)])**_

model.compile(optimizer = gctf.optimizers.adam(),
              loss = 'sparse_categorical_crossentropy',
              metrics=['acc'])

**Comment by lg2578** [comment]
with a random seed =3, I run the fashion-mnist demo,the ruslt is:
Type                   Execution time    Accuracy      Loss
-------------------  ----------------  ----------  --------
Model without gctf:           28.0661    0.886354  0.310736
Model with gctf               29.7088    0.884479  0.319094

tf version is 2.5.1
why it happens?
